olp-cpp-sdk-dataservice-read/src/ReleaseDependencyResolver.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include "ReleaseDependencyResolver.h"
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <utility>
+
+namespace {
+constexpr auto kQuadTreeDepth = 4;
+}  // namespace
+
+namespace olp {
+namespace dataservice {
+namespace read {
+
+cache::KeyValueCache::KeyListType
+ReleaseDependencyResolver::GenerateKeysToRelease(
+    const client::HRN& catalog, const std::string& layer_id,
+    const int64_t& version, const client::OlpClientSettings& settings,
+    const TileKeys& tiles) {
+  auto process_tiles =
+      ReleaseDependencyResolver(catalog, layer_id, version, settings);
+  return process_tiles.GenerateAndMoveKeysToRelease(tiles);
+}
+
+ReleaseDependencyResolver::ReleaseDependencyResolver(
+    const client::HRN& catalog, const std::string& layer_id,
+    const int64_t& version, const client::OlpClientSettings& settings)
+    : layer_id_(layer_id),
+      version_(version),
+      cache_(settings.cache),
+      data_cache_repository_(catalog, settings.cache),
+      partitions_cache_repository_(catalog, settings.cache),
+      repository_(catalog, settings),
+      quad_trees_with_protected_tiles_(),
+      keys_to_release_() {}
+
+cache::KeyValueCache::KeyListType&&
```

Comment:
Why do you return a rvalue when you can simply return the cache::KeyValueCache::KeyListType which is the same except you are not forced to use std::move.
Also the name of the method will get much shorter: `GetKeysToRelease`